src/js/popup/popup.js

```diff
@@ -1,43 +1,117 @@
+import * as Sentry from '@sentry/browser'
+import {sendEvent} from '../analytics/event'
+import {NotesPopup} from './notesPopup'
+import {getSyncedSetting} from '../serviceWorker/utils'
 import {sleep} from '../utils/utils'
 
-const saveButton = document.getElementById('saveKeyButton')
-const analyzeButton = document.getElementById('analyzeButton')
-const stalledWorkCheckbox = document.getElementById('stalledWorkToggle')
-const todoistCheckbox = document.getElementById('todoistOptions')
 
+/**
+ * Handles all interactions with the popup.html file including saving settings
+ * and displaying the correct section. Functionality for saving notes and the OpenAI token are also included.
+ * @class
+ */
+export class Popup {
+  constructor() {
+    this.saveButton = document.getElementById('saveKeyButton')
+    this.analyzeButton = document.getElementById('analyzeButton')
+    this.stalledWorkCheckbox = document.getElementById('stalledWorkToggle')
+    this.todoistCheckbox = document.getElementById('todoistOptions')
+    this.changelogButton = document.getElementById('changelog')
 
-export async function setOpenAIToken(){
+    this.saveButton.addEventListener('click', async () => {
+      await this.saveButtonClicked()
+    })
+    this.changelogButton.addEventListener('click', async () => {
+      await chrome.action.setBadgeText({text: ''})
+    })
```

Comment:
Can these be structured similar to how the DOMContentLoaded callback is passed through?